src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -476,14 +491,14 @@ Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClus
 
 Instance::~Instance()
 {
-    ModeBaseAliasesInstances.erase(this);
+    UnregisterInstance(this);
```

Comment:
If the instances are static, and the list itself is static, what guarantees the destructors of the instances will run before the list destructor?  The latter will crash if the list is not empty.. and in fact I think some of the CI failures on this PR are precisely that.

Should we have a Shutdown() on Instance that applications are responsible for calling?

Reply:
Is there an emberAf shutdown callback function where the user can call this shutdown command?

Static objects should be deconstructed in the reverse order that they where constructed so it shouldn't be an issue since the list should be constructed before the Instances. 

The problem is that the instances in the examples are created on the heap. So they are never deconstructed. We need the application to destroy these objects at shutdown.

Reply:
> Is there an emberAf shutdown callback function where the user can call this shutdown command?

Shutdown happens under application control to start with, right?  Apps call Server::Shutdown.  So they could clean up before doing that?  Or we could do it in Server::Shutdown itself, perhaps.

> Static objects should be deconstructed in the reverse order that they where constructed

Yes.

> so it shouldn't be an issue since the list should be constructed before the Instances.

Absolutely nothing guarantees this, right?  The list is in one translation unit.  The instances are in another.  Their static constructors do not have a defined ordering.

Reply:
In the Linux all-clusters-app, I moved the `AppicationExit()` call to the `ChipLinuxAppMainLoop()` so it can be called before `Server::Shutdown()`. This makes more sense as this Init and Exit pair is called in reverse order with all the other init and de-init methods.

The `AppicationExit()` method calls shutdown methods for all the delegates. These destroy the instance then the delegate. The changes are in [this commit](https://github.com/project-chip/connectedhomeip/pull/28163/commits/d9cab5435caa03b833cc02ca9f005006d68f0321).